# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerTestUtils.h
# Commit: fa2538a4e354
# Full Hash: fa2538a4e354c1d9841b2a611def5843c1e383a6
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-11-05 16:14:41
# Description:
#   Bug 1927247 - Regenerate Client Id for ServiceWorkers on termination. r=dom-worker-reviewers,webidl,smaug
#   
#   Bug 1544232 changed it so ServiceWorker globals used a ClientInfo and
#   Client Id created by the ServiceWorkerPrivate rather than creating a
#   random client id.  This allows the ServiceWorkerManager to reliably map
# ==============================================================================

diff -r 18af0608f7f1 -r fa2538a4e354 dom/workers/WorkerTestUtils.h
--- a/dom/workers/WorkerTestUtils.h	Tue Nov 05 05:18:37 2024 +0000
+++ b/dom/workers/WorkerTestUtils.h	Tue Nov 05 06:26:05 2024 +0000
@@ -7,12 +7,17 @@
 #ifndef mozilla_dom_WorkerTestUtils__
 #define mozilla_dom_WorkerTestUtils__
 
+#include "nsStringFwd.h"
+
 namespace mozilla {
 
 class ErrorResult;
+class GlobalObject;
 
 namespace dom {
 
+class WorkerTestCallback;
+
 /**
  * dom/webidl/WorkerTestUtils.webidl defines APIs to expose worker's internal
  * status for glass-box testing. The APIs are only exposed to Workers with prefs
@@ -35,6 +40,17 @@
                                            ErrorResult& aErr);
 
   static bool IsRunningInBackground(const GlobalObject&, ErrorResult& aErr);
+
+  static void HoldStrongWorkerRefUntilMainThreadObserverNotified(
+      const GlobalObject&, const nsACString& aTopic, ErrorResult& aErr);
+
+  MOZ_CAN_RUN_SCRIPT static void BlockUntilMainThreadObserverNotified(
+      const GlobalObject&, const nsACString& aTopic,
+      WorkerTestCallback& aWhenObserving, ErrorResult& aErr);
+
+  static void NotifyObserverOnMainThread(const GlobalObject&,
+                                         const nsACString& aTopic,
+                                         ErrorResult& aErr);
 };
 
 }  // namespace dom
